edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudProxy.cs

```diff
@@ -126,7 +126,7 @@ public async Task<IMessage> GetTwinAsync()
             {
                 using (Metrics.TimeGetTwin(this.clientId))
                 {
-                    Twin twin = await this.client.GetTwinAsync();
+                    Twin twin = await this.client.GetTwinAsync().TimeoutAfter(this.sdkWaitTimeSeconds);
                     Events.GetTwin(this);
```

Comment:
this uses the old TimeoutAfter version which would throw a timeout exception instead of the specific